### DISCUSSION TITLE
How to create seed and targets files in sbml format?

### DISCUSSION THREAD

**Discussion by brunorbqi** [bioasp/meneco#27] [Q&A] [answered]
Hello ! How do I create the seeds and target files in sbml format ? On github page has exemples but I was unable to download.

**Comment by sthiele** [comment, marked as answer]
SBML are XML files that have a section `listOfSpecies`, there you can list all your seed metabolites (**see below**). For more info about SBML see [https://sbml.org/documents/](https://sbml.org/documents/).

```xml
<?xml version="1.0" encoding="UTF-8"?>
<sbml xmlns="http://www.sbml.org/sbml/level2" level="2" version="1">
<model id="EsiCyc">
  <listOfCompartments>
    <compartment id="CCO__45__EXTRACELLULAR"/>
    <compartment id="CCO__45__CYTOSOL"/>
    <compartment id="CCO__45__OUT"/>
  </listOfCompartments>
  <listOfSpecies>
    <species id="Acceptor" compartment="CCO__45__CYTOSOL"/> 
    <species id="ACP" compartment="CCO__45__CYTOSOL"/> 
    <species id="ADP" compartment="CCO__45__CYTOSOL"/>
    <species id="AMMONIA" compartment="CCO__45__CYTOSOL"/>
    <species id="AMP" compartment="CCO__45__CYTOSOL"/>
    <species id="ATP" compartment="CCO__45__CYTOSOL"/>
    <species id="SULFATE" compartment="CCO__45__CYTOSOL"/>
    <species id="THIAMINE" compartment="CCO__45__CYTOSOL"/>
    <species id="UDP" compartment="CCO__45__CYTOSOL"/>
    <species id="Vitamins__45__B12" compartment="CCO__45__CYTOSOL"/>
    <species id="COB__45__I__45__ALAMIN" compartment="CCO__45__CYTOSOL"/>
    <species id="WATER" compartment="CCO__45__CYTOSOL"/>
    <species id="ZN__43__2" compartment="CCO__45__CYTOSOL"/>
    </listOfSpecies>
  </model>
</sbml>
```

**Github tip:**
If you navigate to the SBML files in this repository like [here](https://github.com/bioasp/meneco/blob/next/Ectodata/seeds.sbml), then you can press the **Raw** button to access the *raw* files like [https://raw.githubusercontent.com/bioasp/meneco/next/Ectodata/seeds.sbml](https://raw.githubusercontent.com/bioasp/meneco/next/Ectodata/seeds.sbml).
![image](https://user-images.githubusercontent.com/1641315/150868011-76144f28-1b92-4008-b116-3383c2afbc3f.png)

Also see the samples section of the [README](https://github.com/bioasp/meneco#samples)

**Comment by cfrioux** [comment]
In addition, there exists a [solution](https://metage2metabo.readthedocs.io/en/latest/command.html#m2m-seeds) to automatise the process by providing a simple text file with a list of encoded identifiers that will be turned into a SBML file.
e.g. M_glucose_c for glucose extracellular.

As an example, the content below:

```
M_glucose_c
M_water_c
```

will generate the following seed file:

```
<?xml version="1.0" encoding="UTF-8"?>
<sbml xmlns="http://www.sbml.org/sbml/level2" level="2" version="1">
<model id="seeds">
  <listOfCompartments>
    <compartment id="c"/>
  </listOfCompartments>
  <listOfSpecies>
    <species id="M_glucose_c" name="glucose" compartment="c"/> 
    <species id="M_water_c" name="water" compartment="c"/> 
    </listOfSpecies>
  </model>
</sbml>
```

**Comment by brunorbqi** [comment]
Hello !!!
I have installed the  m2m with the following command line:
pip install Metage2Metabo

And executed as follows:
m2m seeds -o (my output directory) --metabolites seeds.txt
And got the following error:
m2m seeds -o
Biologia_de_Sistemas/babortus_2.0/Finalizacao_do_Modelo_B_abortus/Acabando
--metabolites seeds.txt
Traceback (most recent call last):
  File "/home/brunobqi/anaconda3/bin/m2m", line 5, in <module>
    from metage2metabo.__main__ import main
  File
"/home/brunobqi/anaconda3/lib/python3.7/site-packages/metage2metabo/__init__.py",
line 15, in <module>
    from metage2metabo import m2m, m2m_analysis
  File
"/home/brunobqi/anaconda3/lib/python3.7/site-packages/metage2metabo/m2m_analysis/__init__.py",
line 16, in <module>
    from metage2metabo.m2m_analysis.graph_compression import
powergraph_analysis, check_oog_jar_file
  File
"/home/brunobqi/anaconda3/lib/python3.7/site-packages/metage2metabo/m2m_analysis/graph_compression.py",
line 25, in <module>
    from bubbletools import convert
ModuleNotFoundError: No module named 'bubbletools'

Em ter., 25 de jan. de 2022 às 03:49, Clémence Frioux <
***@***.***> escreveu:

**Comment by sthiele** [comment]
Looks like you're missing dependencies. You should report this issue at the [metage2metabo issue tracker](https://github.com/AuReMe/metage2metabo/issues).
As a quick fix you can try installing the missing [bubbletools package](https://github.com/aluriak/bubble-tools).
```
pip install bubbletools
```
